### ISSUE TITLE
[Bug]: provider: (#39255) - Not allowing empty role_arn under assume_role block

### ISSUE THREAD

**Issue by devdootkim** [hashicorp/terraform-provider-aws#39306]
### Terraform Core Version

1.0.10 , 1.9.2

### AWS Provider Version

v5.67.0

### Affected Resource(s)

Terraform provider makes error with empty value of role_arn which is under assume_role block after merging the PR provider: (#39255) 

### Expected Behavior

Two types of terraform apply environment.
1. Command based apply 
2. By using Atlantis system

For type 2.
assume_role block is uesd for Atlantis system.
The system gives role_arn value.
There is no issue.

For type 1.
Command based apply is using locally assumed aws role.
No need to assume role again via terraform provider so we made role_arn = "" (empty value).

By v5.66.0, There is no issue with this process.

### Actual Behavior

Type 1, command based apply, 
terraform provider does not allow empty value of role_arn.

```
│ Error: Missing required argument
│ 
│   with provider["registry.terraform.io/hashicorp/aws"],
│   on provider.tf line 4, in provider "aws":
│    4: provider "aws" {
│ 
│ The argument "role_arn" is required, but no definition was found.
```

### Relevant Error/Panic Output Snippet

```shell
Initializing modules...

Initializing the backend...

Initializing provider plugins...
- terraform.io/builtin/terraform is built in to Terraform
- Reusing previous version of hashicorp/aws from the dependency lock file
- Using previously-installed hashicorp/aws v5.67.0
```

```
│ Error: Missing required argument
│ 
│   with provider["registry.terraform.io/hashicorp/aws"],
│   on provider.tf line 4, in provider "aws":
│    4: provider "aws" {
│ 
│ The argument "role_arn" is required, but no definition was found.
```

### Terraform Configuration Files

```
provider "aws" {
  assume_role {
    role_arn     = var.assume_role_arn
  }
}
variable "assume_role_arn" {
  description = ""
  default     = ""
}
```

### Steps to Reproduce

terraform init
```
Initializing modules...

Initializing the backend...

Initializing provider plugins...
- terraform.io/builtin/terraform is built in to Terraform
- Reusing previous version of hashicorp/aws from the dependency lock file
- Using previously-installed hashicorp/aws v5.67.0
```

terraform plan
(with below provider config)
```
provider "aws" {
  assume_role {
    role_arn     = var.assume_role_arn
  }
}
variable "assume_role_arn" {
  description = ""
  default     = ""
}
```

### Would you like to implement a fix?

None

**Comment by devdootkim** [comment]
Workaround works with terraform 1.9.2
https://github.com/cloudfoundry/bosh-bootloader/pull/620/commits/bd8bd524c68856f315cabf9d9a94c95d85de4c01
```
│ Warning: Missing required argument
│ 
│   with provider["registry.terraform.io/hashicorp/aws"],
│   on provider.tf line 1, in provider "aws":
│    1: provider "aws" {
│ 
│ The argument "role_arn" is required, but no definition was found.
│ 
│ This will be an error in a future release.
```

Not work with 1.0.10
```
│ Error: Missing required argument
│ 
│   with provider["registry.terraform.io/hashicorp/aws"],
│   on provider.tf line 4, in provider "aws":
│    4: provider "aws" {
│ 
│ The argument "role_arn" is required, but no definition was found.
```

**Comment by devdootkim** [comment]
WorkAround success with older terraform 

Terraform version 1.0.10
Provider version v5.67.0

```
provider "aws" {

  dynamic "assume_role" {
    for_each = var.assume_role_arn != "" ? [1] : []
    content {
      role_arn     = var.assume_role_arn
    }
  }
}
variable "assume_role_arn" {
  default     = ""
}
```

**Comment by YorickH** [comment]
Duplicate of https://github.com/hashicorp/terraform-provider-aws/issues/39296 ?

**Comment by devdootkim** [comment]
> Duplicate of #39296 ?

Yes the same issue.
I add another workaround for this issue.

**Comment by gdavison** [comment]
This will be resolved by #39328. We're planning to release a bug fix update this Monday, 16 September.

In addition to the workarounds listed above, you can also set the value of `var.aws_assume_role` to `null` instead of `""`. For example:

```terraform
variable "var.aws_assume_role" {
  type    = string
  default = null
}
```
